### PR TITLE
Changes accounting for new repo_config table

### PR DESCRIPTION
**Pull request by tapishr**
## Summary by CodeRabbit

- **New Features**
	- Enhanced repository fetching to personalize data based on user sessions.
	- Added user authentication check before updating repository configurations.
- **Bug Fixes**
	- Improved error handling in the repository list to ensure stability.

### CODE REVIEW

File: utils/db/repos.ts [hunk 1]
```diff
@@ -1,17 +1,44 @@
...
 
-export const getRepos = async (allRepos: RepoIdentifier[]) => {
+export const getRepos = async (allRepos: RepoIdentifier[], session: Session) => {
+	const userId = session.user.id;
+	if (!userId) {
+		console.error("[getRepos] No user id in session", session);
+		return {
+			repos: [],
+			failureRate: 1.01
+		};
```
[the remark this answers is not in the document]
Comment (tapishr) [reply utils/db/repos.ts:13]: @avikalpg - is 1.01 correct or 1.0 ?

File: utils/db/repos.ts [hunk 2]
```diff
@@ -1,17 +1,44 @@
+import { Session } from 'next-auth';
```
Comment (avikalpg) [inline utils/db/repos.ts:1]:
```suggestion
import type { Session } from 'next-auth';
```

Low LOGAF

File: utils/db/repos.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (avikalpg) [reply utils/db/repos.ts:13]: 1 (or 100%) means that all the promises (each promise corresponding to one repository of the user) have failed. We have used 1.01 (101%) as a way to communicate that there were no promises to process in the first place.

This distinction will help us get to the root cause while debugging faster.

**Review by avikalpg — APPROVED** [review]

File: pages/api/setRepoConfig.ts [hunk 3]
```diff
@@ -22,7 +22,7 @@ const setRepoConfigHandler = async (
 ) => {
 	// check auth
 	const session = await getServerSession(req, res, authOptions);
-	if (!session) {
+	if (!session || !session.user.id) {
```
Comment (avikalpg) [inline pages/api/setRepoConfig.ts:25]:
```suggestion
	if (!session?.user.id) {
```

**Review by avikalpg — APPROVED** [review]

**Pull request merged**